packages/blade/src/utils/useBreakpoint/useBreakpoint.ts

```diff
@@ -0,0 +1,144 @@
+              } else if (breakpointsArray[index - 1]) {
+                mediaQuery = `screen and (min-width: ${
+                  breakpointsArray[index - 1][1] + 1
+                }px) and (max-width: ${breakpointSize}px)`;
+              } else {
+                mediaQuery = `screen and (max-width: ${breakpointSize}px)`;
+              }
+
+              return [breakpointTokenName, breakpointSize, mediaQuery];
+            },
+          )
+        : []) as [keyof Breakpoints, number, string][],
+    [breakpoints, supportsMatchMedia],
+  );
+
+  const getMatchedDeviceType = useCallback((matchedBreakpoint: Breakpoint): DeviceType => {
+    let matchedDeviceType: DeviceType = 'mobile';
+    if (typeof navigator !== 'undefined' && navigator.product === 'ReactNative') {
+      // react-native
+      matchedDeviceType = 'mobile';
+    } else if (typeof document !== 'undefined') {
+      // browser
+      if (matchedBreakpoint && ['xs', 's', 'm'].includes(matchedBreakpoint)) {
+        // tablet is also categorised as mobile
+        matchedDeviceType = 'mobile';
+      } else {
+        matchedDeviceType = 'desktop';
+      }
+    } else if (typeof process !== 'undefined') {
+      // node
+      //@TODO: Check for useragent for node
+      matchedDeviceType = 'desktop';
+    }
+    return matchedDeviceType;
+  }, []);
+
+  const getMatchedBreakpoint = useCallback(
+    (event?: MediaQueryListEvent): Breakpoint => {
+      const matchedBreakpoint =
+        breakpointsTokenAndQueryCollection.find(([_, __, mediaQuery]) => {
+          if (event?.media === mediaQuery) {
+            return true;
+          }
+          if (window.matchMedia(mediaQuery).matches) {
+            return true;
+          }
+          return false;
+        })?.[0] ?? undefined;
+
+      return matchedBreakpoint;
+    },
+    [breakpointsTokenAndQueryCollection],
+  );
+
+  const [breakpointAndDevice, setBreakpointAndDevice] = useState(() => {
+    const matchedBreakpoint = getMatchedBreakpoint();
+    const matchedDeviceType = getMatchedDeviceType(matchedBreakpoint);
+    return {
+      matchedBreakpoint,
+      matchedDeviceType,
+    };
+  });
+
+  useEffect(() => {
+    if (!supportsMatchMedia) {
+      return undefined;
+    }
+
+    const handleMediaQueryChange = (event: MediaQueryListEvent): void => {
+      setBreakpointAndDevice((prevBreakpointAndDevice) => {
+        const matchedBreakpoint = getMatchedBreakpoint(event);
+        const matchedDeviceType = getMatchedDeviceType(matchedBreakpoint);
+        if (
+          prevBreakpointAndDevice.matchedBreakpoint !== matchedBreakpoint ||
+          prevBreakpointAndDevice.matchedDeviceType !== matchedDeviceType
+        ) {
+          return { matchedBreakpoint, matchedDeviceType };
+        }
+        return prevBreakpointAndDevice;
+      });
+    };
+
+    const mediaQueryInstances = breakpointsTokenAndQueryCollection.map(([_, __, mediaQuery]) =>
+      window.matchMedia(mediaQuery),
+    );
```

Comment:
the mediaquery event listener is available on mediaQuery instances and not `window` so we need to iterate over all the breakpoints we have and register instance for each and store them as `mediaQueryInstances` so we can later unregister all of them.

More details about [MediaQuery vs window  `addEventListeners`](https://webdevetc.com/blog/matchmedia-events-for-window-resizes/#windowmatchmedia-to-the-rescue)